.NET/Microsoft.Recognizers.Text.DataDrivenTests/TestHelpers.cs

```diff
@@ -0,0 +1,313 @@
+                case "CustomNumberModel":
+                    return GetCustomModelFor(language);
+            }
+            return null;
+        }
+
+        public static IExtractor GetExtractor(this TestContext context)
+        {
+            var language = TestUtils.GetCulture(context.FullyQualifiedTestClassName);
+            var extractorName = TestUtils.GetExtractorName(context.TestName);
+            switch (language)
+            {
+                case Culture.English:
+                    return GetEnglishExtractor(extractorName);
+                case Culture.Spanish:
+                    return GetSpanishExtractor(extractorName);
+            }
+            return null;
+        }
+
+        public static IDateTimeParser GetDateTimeParser(this TestContext context)
+        {
+            var language = TestUtils.GetCulture(context.FullyQualifiedTestClassName);
+            var parserName = TestUtils.GetParserName(context.TestName);
+            switch (language)
+            {
+                case Culture.English:
+                    return GetEnglishParser(parserName);
+                case Culture.Spanish:
+                    return GetSpanishParser(parserName);
+            }
+            return null;
+        }
+
+        public static IExtractor GetEnglishExtractor(string extractorName)
+        {
+            switch (extractorName)
+            {
+                case "Date":
+                    return new BaseDateExtractor(new EnglishDateExtractorConfiguration());
+                case "Time":
+                    return new BaseTimeExtractor(new EnglishTimeExtractorConfiguration());
+                case "DatePeriod":
+                    return new BaseDatePeriodExtractor(new EnglishDatePeriodExtractorConfiguration());
+                case "TimePeriod":
+                    return new BaseTimePeriodExtractor(new EnglishTimePeriodExtractorConfiguration());
+                case "DateTime":
+                    return new BaseDateTimeExtractor(new EnglishDateTimeExtractorConfiguration());
+                case "DateTimePeriod":
+                    return new BaseDateTimePeriodExtractor(new EnglishDateTimePeriodExtractorConfiguration());
+                case "Duration":
+                    return new BaseDurationExtractor(new EnglishDurationExtractorConfiguration());
+                case "Holiday":
+                    return new BaseHolidayExtractor(new EnglishHolidayExtractorConfiguration());
+                case "Set":
+                    return new BaseSetExtractor(new EnglishSetExtractorConfiguration());
+                case "Merged":
+                    return new BaseMergedExtractor(new EnglishMergedExtractorConfiguration(), DateTimeOptions.None);
+                case "MergedSkipFromTo":
+                    return new BaseMergedExtractor(new EnglishMergedExtractorConfiguration(), DateTimeOptions.SkipFromToMerge);
+            }
+            return null;
```

Comment:
I'd rather throw an exception. It makes it easier to immediately see where the error came from. Otherwise a nullpointerexception could come from multiple places. Or null checks need to be spread throughout the test code.

Reply:
Done. All unresolved extractors/parsers or models will throw an exception instead